lib/src/data_table_2.dart

```diff
@@ -31,7 +31,8 @@ class DataColumn2 extends DataColumn {
       super.numeric = false,
       super.onSort,
       this.size = ColumnSize.M,
-      this.fixedWidth});
+      this.fixedWidth,
+      this.alignment});
```

Comment:
I envisioned the change more limited and in the direction of `invisibleSortArrowsTake`. I don't think cell alignment in the header is best solved by integrating it into the widget, I think it is better be solved by developers determining alignment by putting alignment widgets in header's label. Besides this issues presents challenges in regard to handling `numeric` kind of columns which implicitly adds right alignment

Reply:
All tests passed now.


> I envisioned the change more limited and in the direction of `invisibleSortArrowsTake`. I don't think cell alignment in the header is best solved by integrating it into the widget, I think it is better be solved by developers determining alignment by putting alignment widgets in header's label. Besides this issues presents challenges in regard to handling `numeric` kind of columns which implicitly adds right alignment

Are you thinking about a flag in the DataColunm2 that will add a SizedBox on the if the flag is true?
If this is the case this mean asking the developer to set the flag to true AND use the Align in the label.
Not sur it worth the addition.

I think the only use case I can imagine at this point to add the space if when we want to align the header in the middle.  So I opted to add a nice helper extension to the library.

My last change is providing an extension(probably at the right place) and exposing a static variable => sortArrowSpaceWidth 

This way no change in the API! 

At the end, just exposing the **sortArrowSpaceWidth** can be sufficient, but adding a little helper can be cool.

Reply:
The original issue is about invisible arrows taking space. When you tried to center the header using custom alignment widgets in the label of a data column this invisible arrow messed it - is that right? To sort this issues what is needed is a simple bool file in the DataTable2, e.g. `invisibleSortArrowTakesSpace`.

I don't think adding the explicit logic for alignment into the column/table is a good idea:

- It creates implicit dependencies which are hard to spot and can lead to side effects (numeric column)
- It violates the general pricinple Flutter uses, you can rarely find alignment properties in the framework to control the alignment of child widgets (just like for the padding). Flutter suggests to use dedicated widgets (aka aggressive composability). As long as using Align or any other widgets in headers to achieve the required look is OK there's no need in alignment control by the widget (besides already inherited `numeric` behaviour)

Reply:
If you look at the latest code I removed the alignment property.  And all the logic from the core for the alignment like you suggested.

The logic remain in an extension (a utility, if you wish)

Reply:
@maxim-saplin 
If the user use a Center or Align widget for the title, this will push the sort arrow to the end of the Cell, and this is not good.

I look at the DataTable implementation in the framework and this is the same, so I'm almost convince this is a bug (according to the material specification...the arrow need to be close to the header text).

In the mean time, we can solve the issue in the library and propose, I'll propose a PR to the framework as well.

Reply:
I am confused at this stage what is the problem that we are trying to sort out :)

Check the changes, there's a lot of code, increased API surface and complexity for the users of the package and I can't see benefits.

Reply:
@maxim-saplin 
If you look at the latest commit ([47343d7](https://github.com/maxim-saplin/data_table_2/pull/256/commits/47343d79bce5fb9441a2d16b3962795375cc4ec5)), there is no changes to the API surface, and in he code!  Except, exposing 1 global variable and add an Extension (that is optional).  So I don't understand why you mentioned there is a lot of code changes :-)

My second point in my comments is that even if we are trying to provide helper to the user solving the issue around centring the label properly, and it work (I'm currently using it).  But there is another challenge (that exist prior to these changes as well), a side effect of using a Center or Align widget for the label.

Letting the user using a Center or an Align widget will cause a side effect of expanding the header cell and having the sorting arrow now located at the end...see here:
<img width="241" alt="image" src="https://github.com/maxim-saplin/data_table_2/assets/39314023/558709ef-ee17-4203-b114-b37d0a2aaf2b">

Instead of => 
<img width="239" alt="image" src="https://github.com/maxim-saplin/data_table_2/assets/39314023/5b8cf3f6-4b29-46f0-a362-b3d67c084401">


I can understand this can be confusing to take about the location of the sorting arrow, but the side effect is important when we are talking about centring the label.  If there is no solution provided other than giving the space to use before the label.  This will lead letting the user deal this kind of scenario when you are not using cell borders, and this is very confusing about what is column sorted:
<img width="207" alt="image" src="https://github.com/maxim-saplin/data_table_2/assets/39314023/459c0d7a-cffa-4adf-8d67-a9ce833eba7a">

Reply:
When I looked at the original source code in the DataTable (from the framework), this is not even possible to center the label.  Because the label is not wrap in a Flexible widget (like the code in the library)

```
 label = Row(
      textDirection: numeric ? TextDirection.rtl : null,
      children: <Widget>[
        label,
        if (onSort != null)
          ...<Widget>[
            _SortArrow(
              visible: sorted,
              up: sorted ? ascending : null,
              duration: _sortArrowAnimationDuration,
            ),
            const SizedBox(width: _sortArrowPadding),
          ],
      ],
    );
```